skills_ml/ontologies/esco.py

```diff
@@ -0,0 +1,78 @@
+from .base import Competency, Occupation, CompetencyOntology
+import logging
+import csv
+import requests
+
+lang = '&language=en'
+api_tax = 'https://ec.europa.eu/esco/api/resource/taxonomy?uri=http://data.europa.eu/esco/concept-scheme/isco&language=en'
+
+
+class Esco(CompetencyOntology):
+    def __init__(self):
+        super().__init__()
+        self.is_built = False
+        self.competency_framework.name = 'esco_skills'
+        self.competency_framework.description = 'ESCO Knowledge, Skills/Competencies'
+        self._build()
+    
+    def _build(self):
+        if not self.is_built:
+            reqTax = requests.get(api_tax)
+            getConcepts = reqTax.json()['_links']['hasTopConcept']
+            for concept in getConcepts:
+                self._conceptProcessing(concept['href'])
+        else:
+            logging.warning('ESCO Ontology is already built!')
+
+    def _conceptProcessing(href):
```

Comment:
All of the instance methods still need the `self` at the start of their signatures as I'm continuing to get errors like this:
```
   self._conceptProcessing(concept['href'])
TypeError: _conceptProcessing() takes 1 positional argument but 2 were given`
```